snapcraft/tests/plugins/test_ruby.py

```diff
@@ -0,0 +1,227 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2015-2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import platform
+from unittest import mock
+
+from testtools.matchers import Equals, HasLength
+
+import snapcraft
+from snapcraft import tests
+from snapcraft.plugins import ruby
+
+
+class RubyPluginTestCase(tests.TestCase):
+
+    def setUp(self):
+        super().setUp()
+
+        class Options(snapcraft.ProjectOptions):
+            source = '.'
+            ruby_version = '2.4.0'
+            gems = []
+            use_bundler = False
+
+        self.options = Options()
+        self.project_options = snapcraft.ProjectOptions()
+
+    def test_schema(self):
+        schema = ruby.RubyPlugin.schema()
+        expected_use_bundler = {
+            'type': 'boolean',
+            'default': False,
+        }
+        expected_ruby_version = {
+            'type': 'string',
+            'default': '2.4.0'
+        }
+        expected_gems = {
+            'type': 'array',
+            'minitems': 1,
+            'uniqueItems': True,
+            'items': {'type': 'string'},
+            'default': [],
+        }
+
+        self.assertThat(schema['properties'], HasLength(3))
+        self.assertDictEqual(expected_use_bundler,
+                             schema['properties']['use-bundler'])
+        self.assertDictEqual(expected_ruby_version,
+                             schema['properties']['ruby-version'])
+        self.assertDictEqual(expected_gems,
+                             schema['properties']['gems'])
+
+    def test_get_pull_properties(self):
+        expected_pull_properties = [
+            'ruby-version',
+            'gems',
+            'use-bundler'
+        ]
+        resulting_pull_properties = ruby.RubyPlugin.get_pull_properties()
+
+        self.assertThat(resulting_pull_properties,
+                        HasLength(len(expected_pull_properties)))
+
+        for property in expected_pull_properties:
+            self.assertIn(property, resulting_pull_properties)
+
+    def test_snap_fileset(self):
+        expected_fileset = (['-include/', '-share/'])
```

Comment:
beware that excluding `share` could lead to automatically excluding license files